Sources/CodeGenerator/Util.swift

```diff
@@ -58,11 +58,55 @@ func loadAPIJSONList() throws -> [JSON] {
     return try apiPaths.map {
         let data = try Data(contentsOf: URL(string: "file://\($0)")!)
         var json = JSON(data: data)
-        json["serviceName"].stringValue = String($0.split(separator: "/").reversed()[2])
+        json["serviceName"].stringValue = serviceNameForApi(apiJSON: json)
         return json
     }
 }
 
+// port of https://github.com/aws/aws-sdk-go-v2/blob/996478f06a00c31ee7e7b0c3ac6674ce24ba0120/private/model/api/api.go#L105
+//
+var stripServiceNamePrefixes: [String] = [
+  "Amazon",
+  "AWS",
+]
+
+var serviceAliases: [String:String] = [
+	"elasticloadbalancing": "ELB",
+	"elasticloadbalancingv2": "ELBV2",
+	"config": "ConfigService"
+]
+
+func serviceNameForApi(apiJSON: JSON) -> String {
+    var serviceNameJSON = apiJSON["metadata"]["serviceAbbreviation"]
+
+    if serviceNameJSON == nil {
+        serviceNameJSON = apiJSON["metadata"]["serviceFullName"]
+    }
+
+    var serviceName = serviceNameJSON.stringValue
+
+    serviceName.trimCharacters(in: .whitespaces)
+
+    // Strip out prefix names not reflected in service client symbol names.
+    for prefix in stripServiceNamePrefixes {
+        serviceName.deletePrefix(prefix)
+    }
+
+    // Replace all Non-letter/number values with space
```

Comment:
Isn't this deleting those characters? Might be best to delete this comment

Reply:
yes. I lifted the comments from the go sdk which replaces non alphas with space, then removes the spaces later. Will change to `// Remove all Non-letter/number values`